Source/WebCore/rendering/updating/RenderTreeUpdaterGeneratedContent.cpp

```diff
@@ -263,4 +263,50 @@ void RenderTreeUpdater::GeneratedContent::removeAfterPseudoElement(Element& elem
     element.clearAfterPseudoElement();
 }
 
+void RenderTreeUpdater::GeneratedContent::updateWritingSuggestionsRenderer(RenderElement& renderer)
+{
+    auto destroyWritingSuggestionsIfNeeded = [&renderer, this]() {
+        if (WeakPtr writingSuggestionsRenderer = renderer.writingSuggestionsRenderer())
+            m_updater.m_builder.destroy(*writingSuggestionsRenderer);
+    };
+    if (!renderer.element()) {
+        destroyWritingSuggestionsIfNeeded();
+        return;
+    }
+
+    if (renderer.element()->writingSuggestion().isEmpty()) {
+        destroyWritingSuggestionsIfNeeded();
+        return;
+    }
+
+    if (!renderer.canHaveChildren())
+        return;
+
+    auto style = renderer.getCachedPseudoStyle({ PseudoId::InternalWritingSuggestions }, &renderer.style());
+    if (!style || style->display() == DisplayType::None) {
+        destroyWritingSuggestionsIfNeeded();
+        return;
+    }
+
+    if (renderer.writingSuggestionsRenderer() && renderer.writingSuggestionsRenderer()->nextSibling())
+        destroyWritingSuggestionsIfNeeded();
+
+    auto newStyle = RenderStyle::clone(*style);
+    if (auto writingSuggestionsRenderer = renderer.writingSuggestionsRenderer()) {
+        writingSuggestionsRenderer->setStyle(WTFMove(newStyle));
+
+        if (auto child = writingSuggestionsRenderer->firstChild())
+            uncheckedDowncast<RenderText>(child)->setText(renderer.element()->writingSuggestion());
```

Comment:
why is it for this downcast to be unchecked?

Reply:
fixed